tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/CompareApplications.scala

```diff
@@ -36,15 +36,19 @@ class CompareApplications(apps: ArrayBuffer[ApplicationInfo],
     var query = ""
     var i = 1
     for (app <- apps) {
-      query += app.generateAppInfo
-      if (i < apps.size) {
-        query += "\n union \n"
+      if (app.allDataFrames.contains(s"appDF_${app.index}")) {
+        query += app.generateAppInfo
```

Comment:
so this really is the same as printAppInfo which the exception of the union vs printing new tables on each iteration.  I don't think in non-compare mode we ever have more then 1 app,b ut seems like we can combine and just have parameter to change behavior.  That might be the same for other functions in this file like job info, etc. so perhaps we can look through all of them and combine where possible

Reply:
lets do this separately